src/vs/workbench/contrib/searchEditor/browser/searchEditor.ts

```diff
@@ -494,6 +502,8 @@ export class SearchEditor extends AbstractTextCodeEditor<SearchEditorViewState>
 				}
 				if (options.focusResults) {
 					this.searchResultEditor.focus();
+				} else {
+					this.focusSearchInput();
```

Comment:
Is this necessary? I believe that `search.searchEditor.focusResults: false` should mean that it should just keep focus wherever it already is.

Reply:
Yes, this part is a workaround for a use case where a PeekViewWidget was open in previous Search Editor results and after a new search is completed, at the widget closure, the ReferencesController moves focus to results, instead of leaving it at Search Editor input. I will add a comment here too.

Reply:
Sorry, I'm a bit confused- do you want the references view to stay open on refresh? And focusing the search input keeps it open?

Reply:
No, no, the references view is being closed on refresh as it should be. I just meant that some callback sequence is fired when closing this view and it moves the focus to the editor results. When I close the references view manually before new search, the focus stays where I want it to be, at the search input.

Reply:
So are you saying that closing the references view is moving the focus to the editor inadvertently? That seems like a bug with the references peek to me. So the sequence without this line is:
1. refresh of search editor closes the peek editor
2. closing the peek editor sets the cursor to where the peek editor was
3. cursor is now where the peek editor was, even if the focus was on the input previously?

Are you sure that this isn't caused by something like the `options.resetCursor`?

Reply:
Well summed up.

After further testing with fresh eyes, it looks like this is the line that moves the focus:
https://github.com/microsoft/vscode/blob/fca05df7d4a9ce627919af8cd9fc3a3f2e14e029/src/vs/editor/contrib/gotoSymbol/browser/peek/referencesController.ts#L110

The usage of `options.resetCursor` moves the cursor pointer within the text area, but does not change the focus:
https://github.com/microsoft/vscode/blob/fca05df7d4a9ce627919af8cd9fc3a3f2e14e029/src/vs/workbench/contrib/searchEditor/browser/searchEditor.ts#L492-L495

The definition of the `closeWidget` method has enabled the editor focusing by default:

https://github.com/microsoft/vscode/blob/fca05df7d4a9ce627919af8cd9fc3a3f2e14e029/src/vs/editor/contrib/gotoSymbol/browser/peek/referencesController.ts#L229-L240

so when I test with the line set to
```
this.closeWidget(false); 
```
the focus stays in the input where I want it to be. But I am afraid of this change, because someone could complain that the exit button of the peek viewer doesn't work as before because it no longer returns focus to the editor.

So I would stay with the `this.focusSearchInput();` call because it looks like a safe compromise.

Reply:
How about changing the line that calls it to:
```typescript
controller?.closeWidget(this.searchResultEditor.hasTextFocus());
```
Then, it will act differently the same if the editor has focus. Otherwise, it will not try to focus the editor.

Reply:
I think I'm a bit confused about what you mean, since that line in `referencesController` is called from 
https://github.com/microsoft/vscode/blob/660018a75f944cd9cc27073eef4fc6918ef7ae23/src/vs/workbench/contrib/searchEditor/browser/searchEditor.ts#L615
which already specifies `focusEditor = false`?

Reply:
Oh, I see in the stack trace: `closeWidget` gets called twice, where the second time gets called with `true` regardless of what the original call had. Let me investigate.